src/corehost/cli/deps_resolver.cpp

```diff
@@ -539,45 +314,63 @@ void deps_resolver_t::resolve_tpa_list(
         {
             add_tpa_asset(entry.asset_name, candidate, &items, output);
         }
-        // Is this entry present in the secondary package cache?
+        // Is an NI image for this entry present in the secondary package cache?
+        else if (entry.to_hash_matched_path(ni_package_cache_dir, &candidate))
```

Comment:
The hash would be for the whole nupkg, so I think it would be present. The hash is really just a unique string that's expected to be identical between the deps file and the cache. A SHA-1 hash of the nupkg is the most convenient way to achieve that, but it's OK if the hash is used to match assets **derived** from the package (i.e. NIs) as well as assets contained directly in the package.